src/diagnostics/radiation_diagnostics.jl

```diff
@@ -39,6 +39,93 @@ add_diagnostic_variable!(
     compute! = compute_rsd!,
 )
 
+###
+# TOA downwelling shortwave radiation (3d)
+###
+compute_rsdt!(out, state, cache, time) =
+    compute_rsdt!(out, state, cache, time, cache.atmos.radiation_mode)
+compute_rsdt!(_, _, _, _, radiation_mode::T) where {T} =
+    error_diagnostic_variable("rsdt", radiation_mode)
+
+function compute_rsdt!(
+    out,
+    state,
+    cache,
+    time,
+    radiation_mode::T,
+) where {T <: RRTMGPI.AbstractRRTMGPMode}
+    nlevels = Spaces.nlevels(axes(state.c))
+    if isnothing(out)
+        return Fields.level(
+            RRTMGPI.array2field(
+                cache.radiation.radiation_model.face_sw_flux_dn,
+                axes(state.f),
+            ),
+            nlevels + half,
+        )
+    else
+        out .= Fields.level(
+            RRTMGPI.array2field(
+                cache.radiation.radiation_model.face_sw_flux_dn,
+                axes(state.f),
+            ),
+            nlevels + half,
+        )
+    end
```

Comment:
@Sbozzolo @charleskawczynski Any idea why this doesn't seem to work on GPU? https://buildkite.com/clima/climaatmos-ci/builds/16180#018d44bc-b6a9-443b-8969-0c6befdbab2d/163-260

Reply:
Hmm, for some reason it's doing scalar indexing. I can checkout the branch to see what's going on